### ISSUE TITLE
Print type when argument is null

### ISSUE THREAD

**Issue by bric3** [mockito/mockito#716]
Sometime when overloaded method are invoked with `null` it is hard to know which method has been invoked.

```
mock.varargs(null);
```

it should print something like

```
mock.varargs((Object) null);
mock.varargs((Object[]) null);
```

**Comment by fdonascimento** [comment]
Hi. I would like to work on this issue. Could anyone please explain how to reproduce this issue?
I was able to reproduce something similar using this:
```

public class ServiceTest {

    @Test
    public void test() {
        Service service = mock(Service.class);
        when(service.execute(null)).thenReturn(1);

        service.execute(null);
        verify(service).execute((Object) null);

    }
}

interface Service {
    int execute(String message);
    int execute(Object object);
}
```

And this gives this message:
```
Wanted but not invoked:
service.execute(null);
-> at services.ServiceTest.test(ServiceTest.java:16)

However, there was exactly 1 interaction with this mock:
service.execute(null);
-> at services.ServiceTest.test(ServiceTest.java:15)
```

Is this situation that you are referring to?

**Comment by fdonascimento** [comment]
@bric3 @mockitoguy Hi!

I was trying to solve this issue, but I have some questions about what is the desired behavior.

1 - Should the type be printed only if the argument is null? We have the same situation with other types. Given the following print `method("toString")`, although the toString can give me the clue of what overloaded method was called it could be more clear if the type was shown explicitly.

2 - What should happen if we have an overloaded method with vargs?
Given the following methods:
```
method (String param1, String param2, String param3)
method(String param1, String... param2)
```

For the first one, it is clear to me that we should have this:
`method((String) null, (String) null, (String) null)`
But what about the second? If we have the same print, we still don't know which overloaded method was invoked.

Same question for when we have these options:
```
method(String[] param)
method(String... param)
```

Given this invocation `method(null)`, for the first one we could print: `method((String[]) null)` 
But what about the second?

**Comment by TimvdLippe** [comment]
This has been implemented for #2311 in #2320
